src/App/Utilities/IconConverter.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Globalization;
+using Bit.Core;
+using Bit.Core.Abstractions;
+using Bit.Core.Enums;
+using Bit.Core.Models.View;
+using Bit.Core.Utilities;
+using Xamarin.Forms;
+
+namespace Bit.App.Utilities
+{
+    public class IconConverter : IValueConverter
+    {
+        private readonly IEnvironmentService _environmentService = ServiceContainer.Resolve<IEnvironmentService>("environmentService");
+
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+
+            var cipher = value as CipherView;
+            var iconImage = GetIconImage(cipher);
+            if (iconImage.Item2 != null)
+            {
+
+                return iconImage.Item2;
+            }
+            else
+            {
+                return iconImage.Item1;
+            }
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Tuple<string, string> GetIconImage(CipherView cipher)
+        {
+            string icon = null;
+            string image = null;
+            switch (cipher.Type)
+            {
+                case CipherType.Login:
+                    var loginIconImage = GetLoginIconImage(cipher);
+                    icon = loginIconImage.Item1;
+                    image = loginIconImage.Item2;
+                    break;
+                case CipherType.SecureNote:
+                    icon = "";
```

Comment:
Are these encoded correctly? Just want to ensure the character is still valid/good. IMO would be more clear to use the unicode char code w/ escape character here explicitly defined, or hex representation of that char code, vs. the actual embedded unicode character (which depending on your editor is pretty hard to decipher/check).